### PR TITLE
fix(instr-mongodb): fix  function patch missing one argument introduced in v6.8.0

### PR DESCRIPTION
**Pull request by david-luna**
…

## Which problem is this PR solving?

`mongodb@6.8.0` introduced a new feature for lazy parsing documents. This feature relies on a new parameter on an internal API that the instrumentation was not passing to the original method resulting in an exception being thrown. Ref: https://github.com/open-telemetry/opentelemetry-js-contrib/issues/2309

this PR contains a fix to make sure the `command` original API gets all parameters when called. It uses `Function.prototype.apply` to call the original function so we do not stump on the same issue if any other param is added.

## Short description of the changes

- use `Function.prototype.apply` to not lose any parameters when calling the original API
- added some extra types
- updated tests since one of them was inspecting data which is not parsed yet

### CODE REVIEW

File: plugins/node/opentelemetry-instrumentation-mongodb/test/mongodb-v5-v6.test.ts [hunk 1]
```diff
@@ -499,12 +499,23 @@ describe('MongoDBInstrumentation-Tracing-v5', () => {
...
         create({
-          responseHook: (span: Span, result: MongoResponseHookInformation) => {
-            span.setAttribute(dataAttributeName, JSON.stringify(result.data));
+          responseHook: (span: Span, result: any) => {
+            const { data } = result;
+            if (data.n) {
+              span.setAttribute('mongodb_insert_count', result.data.n);
+            }
+            // from v6.8.0 the cursor preoperty is not an object but an intance of
+            // `CursoResponse`. We need to use the `toObject` method to be able to inspect the data
```
Comment (pichlermarc) [inline plugins/node/opentelemetry-instrumentation-mongodb/test/mongodb-v5-v6.test.ts:511]:
```suggestion
            // `CursorResponse`. We need to use the `toObject` method to be able to inspect the data
```

File: plugins/node/opentelemetry-instrumentation-mongodb/test/mongodb-v5-v6.test.ts [hunk 2]
```diff
@@ -499,12 +499,23 @@ describe('MongoDBInstrumentation-Tracing-v5', () => {
...
       beforeEach(() => {
         create({
-          responseHook: (span: Span, result: MongoResponseHookInformation) => {
-            span.setAttribute(dataAttributeName, JSON.stringify(result.data));
+          responseHook: (span: Span, result: any) => {
+            const { data } = result;
+            if (data.n) {
+              span.setAttribute('mongodb_insert_count', result.data.n);
+            }
+            // from v6.8.0 the cursor preoperty is not an object but an intance of
```
Comment (pichlermarc) [inline plugins/node/opentelemetry-instrumentation-mongodb/test/mongodb-v5-v6.test.ts:510]:
```suggestion
            // from v6.8.0 the cursor property is not an object but an instance of
```

File: plugins/node/opentelemetry-instrumentation-mongodb/test/mongodb-v5-v6.test.ts [hunk 3]
```diff
@@ -533,7 +543,7 @@ describe('MongoDBInstrumentation-Tracing-v5', () => {
         });
       });
 
-      it('should attach response hook data to the resulting span for find function', done => {
+      it.only('should attach response hook data to the resulting span for find function', done => {
```
Comment (pichlermarc) [inline plugins/node/opentelemetry-instrumentation-mongodb/test/mongodb-v5-v6.test.ts:546]:
```suggestion
      it('should attach response hook data to the resulting span for find function', done => {
```

**Review by pichlermarc — APPROVED** [review]
@david-luna thanks for working on this. The fix looks good, just a few nits from my side. :+1:

**Review by trentm — APPROVED** [review]

**Pull request merged**